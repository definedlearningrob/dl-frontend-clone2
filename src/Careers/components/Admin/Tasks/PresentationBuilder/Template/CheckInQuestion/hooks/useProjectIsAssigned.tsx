import { useMemo } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { isEmpty } from 'lodash-es';

import teamProjectQuery from '@pbl/graphql/student/queries/teamProject';
import projectQuery from '@pbl/graphql/student/queries/project';

import { useRole } from '@shared/hooks/useRole';

export const useProjectIsAssigned = () => {
  const { isStudent } = useRole();
  const { projectId, teamId } = useParams<{ projectId: string; teamId: string }>();

  const properQuery = useMemo(() => (teamId ? teamProjectQuery : projectQuery), [teamId]);

  const queryVariables = useMemo(
    () =>
      teamId
        ? { id: projectId, teamId, track: true, trackPresentation: true }
        : { id: projectId, track: true, trackPresentation: true },
    [teamId]
  );

  const { data, error, loading } = useQuery(properQuery, {
    variables: queryVariables,
    skip: !isStudent,
  });

  if (!isStudent || loading || error) {
    return false;
  }

  return !isEmpty(data?.project?.assignedAt);
};
