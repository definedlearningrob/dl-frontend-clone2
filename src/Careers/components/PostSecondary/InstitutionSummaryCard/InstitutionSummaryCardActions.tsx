import { useTranslation } from 'react-i18next';

import { ReactComponent as LinkIcon } from '@shared/svg/link.svg';
import SharedLink from '@shared/components/Link/Link';
import { formatExternalLink } from '@shared/utils/formatExternalLink';
import { InterestedButton } from '@shared/components/InterestedButton/InterestedButton';

type Props = {
  withoutFavorite?: boolean;
  websiteUrl: string | null;
  isFavorite?: boolean;
  toggleFavorite: () => void;
};

export const InstitutionSummaryCardActions = ({
  withoutFavorite,
  websiteUrl,
  isFavorite,
  toggleFavorite,
}: Props) => {
  const { t } = useTranslation();

  return (
    <>
      {!withoutFavorite && (
        <InterestedButton
          isSelected={isFavorite}
          label={t('postSecondary.institution.favorite')}
          size='sm'
          withLabel={true}
          onClick={toggleFavorite}
        />
      )}
      {websiteUrl && (
        <SharedLink
          Icon={LinkIcon}
          className='text-xs'
          size='sm'
          target='_blank'
          to={{ pathname: formatExternalLink(websiteUrl) }}
          variant='primary-outlined'>
          {t('postSecondary.institution.website')}
        </SharedLink>
      )}
    </>
  );
};
