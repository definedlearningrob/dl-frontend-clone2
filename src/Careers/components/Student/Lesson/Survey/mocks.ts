import lessonInCourseQuery from '@dc/graphql/student/queries/lessonInCourse';
import { CREATE_CAREER_REVIEW_SURVEY_ATTEMPT } from '@dc/graphql/student/mutations/createCareerReviewSurveyAttempt';
import createCareerReviewSurveyAnswersMutation from '@dc/graphql/student/mutations/createCareerReviewSurveyAnswers';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

export const createAttemptSpy = jest.fn();
export const lessonInCourseSpy = jest.fn();

export let createCareerReviewSurveyAnswersSpy = jest.fn();
export let currentCoursesSpy = jest.fn();

export const mocks = [
  {
    request: {
      query: lessonInCourseQuery,
      variables: { courseId: '2', lessonId: '111', track: true },
    },
    result: () => {
      lessonInCourseSpy();

      return {
        data: {
          course: {
            careerName: 'Dental Laboratory Technicians',
            content: {
              checkIns: [],
              items: [],
            },
            hasInstitutionsInStudentState: true,
            id: '2',
            lesson: {
              assignments: [],
              attachments: [],
              careerReviewSurvey: {
                version: 1,
                performed: false,
                questions: [
                  {
                    answer: [],
                    id: '1',
                    options: [
                      {
                        option: 'Yes, definitely',
                        step: 1,
                      },
                      {
                        option: 'Definitely not',
                        step: 2,
                      },
                    ],
                    question: 'Can blind people see their dreams?',
                    type: 'single_choice' as SURVEY_QUESTION_TYPE,
                  },
                  {
                    answer: [],
                    id: '2',
                    options: [
                      {
                        option: 'Very Interested',
                        step: 1,
                      },
                      {
                        option: 'Not at all Interested',
                        step: 2,
                      },
                      {
                        option: 'Other',
                        step: 3,
                      },
                    ],
                    question: 'How interested are you in the Pathway this course is a part of?',
                    type: 'multiple_choice' as SURVEY_QUESTION_TYPE,
                  },
                ],
                __typename: 'CareerReviewSurvey',
              },
              checkInGroups: [],
              checkInQuestions: [],
              description: {
                introduction: null,
                goal: null,
                role: null,
                audience: null,
                situation: null,
                __typename: 'LessonDescription',
              },
              hasPresentation: false,
              id: '609',
              imageUrl: 'http://test.com/image.png',
              name: 'Career Review Survey',
              externalPresentations: [],
              researchLinks: [],
              texts: [],
              type: 'career_review_survey',
              videos: [],
              vocabularies: [],
              __typename: 'Lesson',
            },
            lessons: [
              {
                id: '782',
                step: 4,
                careerReviewSurvey: null,
              },
              {
                id: '609',
                step: 5,
                careerReviewSurvey: {
                  performed: false,
                  version: 1,
                  __typename: 'CareerReviewSurvey',
                },
              },
              {
                id: '275',
                step: 1,
                careerReviewSurvey: null,
              },
              {
                id: '331',
                step: 2,
                careerReviewSurvey: null,
              },
              {
                id: '668',
                step: 3,
                careerReviewSurvey: null,
              },
            ],
            name: 'Dental Laboratory Technician',
            progress: {
              submitted: 10,
              total: 10,
            },
            reviewSurvey: {
              questions: [
                {
                  id: '1',
                },
                {
                  id: '2',
                },
                {
                  id: '3',
                },
                {
                  id: '4',
                },
                {
                  id: '5',
                },
                {
                  id: '6',
                },
                {
                  id: '7',
                },
                {
                  id: '8',
                },
                {
                  id: '9',
                },
                {
                  id: '10',
                },
                {
                  id: '11',
                },
              ],
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: CREATE_CAREER_REVIEW_SURVEY_ATTEMPT,
      variables: { input: { contextType: 'COURSE', contextId: '1' } },
    },
    result: () => {
      createAttemptSpy();

      return {
        data: {
          createCareerReviewSurveyAttempt: {
            careerReviewSurveyAttempt: {
              id: '2',
              status: 'IN_PROGRESS',
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: createCareerReviewSurveyAnswersMutation,
      variables: {
        input: {
          contextId: '1',
          contextType: 'COURSE',
          answers: [
            { questionId: '1', answer: ['Yes, definitely'] },
            { questionId: '2', answer: ['Very Interested'] },
          ],
        },
      },
    },
    result: () => {
      createCareerReviewSurveyAnswersSpy();

      return {
        data: {
          createCareerReviewSurveyAnswers: {
            status: 'success',
          },
        },
      };
    },
  },
  {
    request: {
      query: currentCoursesQuery,
    },
    result: () => {
      currentCoursesSpy();

      return {
        data: {
          currentCourses: {
            id: '1',
            imageUrl: '',
            name: '',
            progress: {
              submitted: 2,
              total: 5,
            },
            status: 'success',
            pathway: {
              name: '',
            },
            thumbnailUrl: '',
            type: 'survey',
          },
        },
      };
    },
  },
];
