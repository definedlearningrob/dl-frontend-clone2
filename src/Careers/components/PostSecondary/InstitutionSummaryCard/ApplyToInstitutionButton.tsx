import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import { NetworkStatus } from '@apollo/client';
import cx from 'classnames';

import { useInstitutionQuery } from '@dc/graphql/student/hooks/useInstitutionQuery';

import { formatExternalLink } from '@shared/utils/formatExternalLink';
import Link from '@shared/components/Link';

type Props = {
  commonAppApplicationUrl: string | null;
  commonAppEnabled: boolean;
  directApplicationURL: string | null;
  onCommonAppApplication: () => void;
};

export const ApplyToInstitutionButton = ({
  commonAppApplicationUrl,
  directApplicationURL,
}: Props) => {
  const { t } = useTranslation();
  const { id } = useParams<{ id: string }>();
  const { networkStatus } = useInstitutionQuery({ id, notifyOnNetworkStatusChange: true });

  const isInstitutionRefetching = networkStatus === NetworkStatus.refetch;

  const commonClasses = cx(
    'button flex items-center justify-center gap-xs rounded-sm',
    'cursor-pointer text-sm font-medium no-underline py-xs px-sm',
    '!text-xs leading-[14px] !gap-xxs border'
  );

  if (commonAppApplicationUrl) {
    return (
      <div className='flex gap-xs xxxl:gap-sm'>
        <Link
          className={commonClasses}
          size='md'
          target='_blank'
          to={{ pathname: formatExternalLink(`${commonAppApplicationUrl}`) }}
          variant='primary'>
          {t('postSecondary.institution.commonAppApplication')}
        </Link>
        <Link
          className={commonClasses}
          size='md'
          target='_blank'
          to={{ pathname: formatExternalLink(`${directApplicationURL}`) }}
          variant='primary-outlined'>
          {t('postSecondary.institution.directApplication')}
        </Link>
      </div>
    );
  }

  return (
    <Link
      disabled={isInstitutionRefetching}
      size='md'
      target='_blank'
      to={{ pathname: formatExternalLink(`${directApplicationURL}`) }}
      variant='primary'>
      {t('postSecondary.institution.directApplication')}
    </Link>
  );
};
