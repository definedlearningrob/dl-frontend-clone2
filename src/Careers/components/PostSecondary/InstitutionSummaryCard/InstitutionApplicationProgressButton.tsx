import { useTranslation } from 'react-i18next';
import { useToggle } from 'react-use';

import { useInstitutionApplication } from '@dc/graphql/student/hooks/useInstitutionApplication';

import Button from '@shared/components/Button/Button';

import { ApplicationDetailsModal } from '../ApplicationDetailsModal/ApplicationDetailsModal';

type Props = {
  applicationId: string;
};

export const InstitutionApplicationProgressButton = ({ applicationId }: Props) => {
  const { t } = useTranslation();
  const [isModalOpen, toggleModalOpen] = useToggle(false);
  const { data } = useInstitutionApplication(applicationId);

  const institutionApplication = data?.institutionApplication;

  return (
    <>
      <Button className='ml-md px-xs py-sm' size='sm' variant='primary' onClick={toggleModalOpen}>
        {t('postSecondary.institution.seeProgress')}
      </Button>
      {institutionApplication && (
        <ApplicationDetailsModal
          isOpen={isModalOpen}
          selectedApplication={institutionApplication}
          onCloseModal={toggleModalOpen}
        />
      )}
    </>
  );
};
