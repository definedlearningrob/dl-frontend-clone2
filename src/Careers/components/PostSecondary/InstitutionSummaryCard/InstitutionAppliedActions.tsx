import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import cx from 'classnames';

import { APPLICATIONS_TYPE } from '@dc/resources/enums';
import { useDeleteInstitutionApplication } from '@dc/graphql/student/hooks/useDeleteInstitutionApplication';

import { ReactComponent as DoneIcon } from '@shared/assets/icons/done.svg';
import Button from '@shared/components/Button/Button';
import SharedIcon from '@shared/components/Icon/Icon';

import { InstitutionApplicationProgressButton } from './InstitutionApplicationProgressButton';

type Props = {
  applicationId: string | null;
  applicationType: APPLICATIONS_TYPE | null;
};

export const InstitutionAppliedActions = ({ applicationId, applicationType }: Props) => {
  const { t } = useTranslation();
  const { id: institutionId } = useParams<{ id: string }>();
  const [deleteInstitutionApplication, { loading }] = useDeleteInstitutionApplication();

  const isCommonAppApplication = applicationType === APPLICATIONS_TYPE.COMMON_APP;
  const isDirectApplication = applicationType === APPLICATIONS_TYPE.DIRECT;

  const shouldDisplayProgressButton = isCommonAppApplication && applicationId;

  const hasAdditionalActions = isDirectApplication || shouldDisplayProgressButton;

  return (
    <div className='border border-neutral-300 p-sm w-fit rounded-sm'>
      <div className={cx('flex items-center gap-xs', { 'mb-sm': hasAdditionalActions })}>
        <div className='bg-primary-200 rounded-md w-fit'>
          <SharedIcon
            className='mb-0 text-primary-500'
            icon={<DoneIcon className='!w-sm !h-sm' />}
            size='sm'
          />
        </div>
        <p className='!mb-0 text-primary-500 text-xs font-medium leading-lg'>
          {t('postSecondary.institution.appliedToCollege')}
        </p>
      </div>
      {shouldDisplayProgressButton && (
        <InstitutionApplicationProgressButton applicationId={applicationId} />
      )}
      {isDirectApplication && (
        <Button
          disabled={loading}
          size='sm'
          variant='primary-outlined'
          onClick={() => deleteInstitutionApplication(applicationId, institutionId)}>
          {t('postSecondary.institution.removeApplication')}
        </Button>
      )}
    </div>
  );
};
