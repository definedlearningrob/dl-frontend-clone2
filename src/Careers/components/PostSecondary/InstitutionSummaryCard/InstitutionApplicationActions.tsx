import { useState } from 'react';

import { APPLICATIONS_TYPE } from '@dc/resources/enums';

import { ApplyToInstitutionButton } from './ApplyToInstitutionButton';
import { InstitutionAppliedActions } from './InstitutionAppliedActions';

type Props = {
  hasApplied: boolean;
  directApplicationURL: string | null;
  commonAppEnabled: boolean;
  commonAppApplicationUrl: string | null;
  applicationId: string | null;
  applicationType: APPLICATIONS_TYPE | null;
};

export const InstitutionApplicationActions = ({
  hasApplied,
  directApplicationURL,
  commonAppEnabled,
  commonAppApplicationUrl,
  applicationId,
  applicationType,
}: Props) => {
  const [hasAppliedToCommonApp, setHasAppliedToCommonApp] = useState(false);
  const hasAppliedToInstitution = hasApplied || hasAppliedToCommonApp;

  if (hasAppliedToInstitution) {
    return (
      <InstitutionAppliedActions applicationId={applicationId} applicationType={applicationType} />
    );
  }

  return (
    <ApplyToInstitutionButton
      commonAppApplicationUrl={commonAppApplicationUrl}
      commonAppEnabled={commonAppEnabled}
      directApplicationURL={directApplicationURL}
      onCommonAppApplication={() => setHasAppliedToCommonApp(true)}
    />
  );
};
