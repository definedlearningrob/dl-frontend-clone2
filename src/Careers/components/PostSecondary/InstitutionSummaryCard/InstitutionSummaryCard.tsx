import { useTranslation } from 'react-i18next';
import { isEmpty } from 'lodash-es';
import { useParams } from 'react-router-dom';
import cx from 'classnames';

import { TInstitution } from '@dc/resources/types';
import useUserInfo from '@dc/hooks/useUserInfo';
import { TStudentInfo } from '@dc/graphql/student/queries/userInfo';
import { useToggleInstitutionFavorite } from '@dc/graphql/student/hooks/useToggleInstitutionFavorite';
import { InstitutionApplicationActions } from '@dc/components/PostSecondary/InstitutionSummaryCard/InstitutionApplicationActions';
import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import { ReactComponent as CommonAppLogo } from '@dc/svg/CommonApp.svg';

import defaultThumbnail from '@shared/assets/images/institution-fallback-image.jpeg';
import { HeaderBox } from '@shared/components/HeaderBox';
import { callToast } from '@shared/components/Toaster/Toaster';

import { InstitutionSummaryCardActions } from './InstitutionSummaryCardActions';

type Props = {
  institution: TInstitution;
  isTeacher?: boolean;
};

export const InstitutionSummaryCard = ({ institution, isTeacher }: Props) => {
  const { t } = useTranslation();

  const { userInfo } = useUserInfo<TStudentInfo | TUserInfo>();

  const isStudent = 'postSecondaryApplicationsEnabled' in userInfo;

  const {
    name,
    imageUrl,
    address,
    sizeType,
    commonAppEnabled,
    type,
    degrees,
    contact,
    commonAppApplicationUrl,
    hasApplied,
    applicationId,
    applicationType,
    isIpeds: isIpedsInstitution,
  } = institution;

  const [toggleFavorite, { loading }] = useToggleInstitutionFavorite();
  const { id } = useParams<{ id: string }>();
  const location = [address.city, address.stateCode].join(', ');
  const size = t(`postSecondary.institution.sizes.${sizeType}`);
  const descriptionKey = sizeType
    ? 'postSecondary.institution.descriptionWithSize'
    : 'postSecondary.institution.description';
  const institutionDegrees = isEmpty(degrees)
    ? t('postSecondary.institution.degreesFallback')
    : degrees.join(', ');
  const isFavorite = institution.isFavorite;
  const handleFavorite = async () => {
    try {
      await toggleFavorite({ institutionId: id });
      !loading && isFavorite
        ? callToast('info', t('student.institutionSearch.institutionRemovedFromFavorite'))
        : callToast('favorite', t('student.institutionSearch.institutionAddedToFavorite'));
    } catch {}
  };

  const checkIfHasPostSecondaryApplicationsEnabled = (userInfo: TUserInfo | TStudentInfo) => {
    if ('postSecondaryApplicationsEnabled' in userInfo) {
      return userInfo.postSecondaryApplicationsEnabled;
    }

    return userInfo.entities.nodes.some(
      ({ settings }) => settings.postSecondaryApplicationsEnabled
    );
  };

  const showCommonAppLogo =
    commonAppEnabled && checkIfHasPostSecondaryApplicationsEnabled(userInfo);

  return (
    <HeaderBox
      actions={
        <InstitutionSummaryCardActions
          isFavorite={isFavorite}
          toggleFavorite={handleFavorite}
          websiteUrl={contact.urlGeneral}
          withoutFavorite={isTeacher}
        />
      }
      fallbackSrc={defaultThumbnail}
      imageTag={showCommonAppLogo && <CommonAppLogo />}
      imageUrl={imageUrl || defaultThumbnail}
      tag={location}
      title={name}>
      {isIpedsInstitution && (
        <p className='mb-base xxxl:mb-md'>
          {t(descriptionKey, {
            name,
            type,
            size,
            areaKind: address.area.kind,
            areaType: address.area.type,
            degrees: institutionDegrees,
          })}
        </p>
      )}
      {isStudent && (
        <div className={cx({ 'mt-sm': !isIpedsInstitution })}>
          <p>{t('postSecondary.institution.beginApplication')}:</p>
          <InstitutionApplicationActions
            applicationId={applicationId}
            applicationType={applicationType}
            commonAppApplicationUrl={commonAppApplicationUrl}
            commonAppEnabled={commonAppEnabled}
            directApplicationURL={contact.urlGeneral}
            hasApplied={hasApplied}
          />
        </div>
      )}
    </HeaderBox>
  );
};
