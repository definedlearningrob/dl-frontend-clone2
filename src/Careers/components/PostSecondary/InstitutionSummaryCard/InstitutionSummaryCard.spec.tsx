import { MockedProvider } from '@apollo/client/testing';
import { screen } from '@testing-library/react';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { InstitutionSummaryCard } from '@dc/components/PostSecondary/InstitutionSummaryCard/InstitutionSummaryCard';
import { INSTITUTION_TYPES } from '@dc/resources/enums';
import cacheConfig from '@dc/graphql/cacheConfig';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { studentInfoMock } from '@dc/tests/mocks/studentMocks';

const institution = {
  id: '861',
  type: INSTITUTION_TYPES.PUB_4,
  name: 'University of Idaho',
  size: 'medium',
  sizeType: 'MEDIUM' as const,
  sizeDescription: 'MORE_THAN_10000' as const,
  cost: null,
  description: '',
  hasApplied: false,
  applicationType: null,
  applicationId: null,
  commonAppApplicationUrl: 'url-for-application',
  imageUrl: 'image-url',
  isFavorite: true,
  maxTeacherEval: 0,
  minTeacherEval: 5,
  thumbnailUrl: 'thumbnail-url',
  address: {
    street: '875 Perimeter Drive MS 2282',
    city: 'Moscow',
    zip: '83844-2282',
    state: 'Idaho',
    stateCode: 'ID',
    area: {
      kind: 'distant',
      type: 'town',
    },
  },
  admissionRate: 0.8138373751783167,
  satMathMin: 490,
  satMathMax: 600,
  satReadingMin: 500,
  satReadingMax: 620,
  actMin: 20,
  actMax: 28,
  studentFacultyRatio: 16,
  commonAppEnabled: true,
  dates: [
    {
      deadlineDate: '2023-03-28',
      decisionType: 'Early Decision',
      term: 'Spring',
    },
    {
      deadlineDate: '2023-04-07',
      decisionType: 'Early Action',
      term: 'Spring',
    },
    {
      deadlineDate: '2023-04-17',
      decisionType: 'Regular Decision',
      term: 'Spring',
    },
    {
      deadlineDate: '2023-10-17',
      decisionType: 'Regular Decision',
      term: 'Fall',
    },
  ],
  degrees: ['certificates', "bachelor's degrees", "master's degrees", 'doctoral degrees'],
  contact: {
    phone: '8888843246',
    urlAdmissions: 'www.uidaho.edu/admissions',
    urlApplications: 'www.uidaho.edu/admissions/apply',
    urlFinancialAid: 'www.uidaho.edu/financialaid',
    urlNetPriceCalculator: null,
    urlGeneral: 'www.uidaho.edu/',
  },
  isIpeds: true,
};

const commonAppEnabledUserInfo = {
  ...studentInfoMock.result.data.userInfo,
  postSecondaryApplicationsEnabled: true,
  commonAppData: {
    ...studentInfoMock.result.data.userInfo.commonAppData,
    hasAccountConnected: true,
  },
};

describe('InstitutionSummaryCard', () => {
  it('should render correctly', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <UserInfoProvider
        value={{
          userInfo: commonAppEnabledUserInfo,
        }}>
        <MockedProvider
          cache={cacheConfig}
          mocks={[
            {
              ...studentInfoMock,
              result: {
                data: {
                  userInfo: commonAppEnabledUserInfo,
                },
              },
            },
          ]}>
          <InstitutionSummaryCard institution={institution} />
        </MockedProvider>
      </UserInfoProvider>,
      {
        initialState: { session: { loginError: {}, user: { type: 'student' } } },
      }
    );

    expect(screen.getByText('University of Idaho')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should always display apply buttons', () => {
    renderWithRouterAndReduxProvider(
      <UserInfoProvider
        value={{
          userInfo: {
            ...studentInfoMock.result.data.userInfo,
            postSecondaryApplicationsEnabled: false,
          },
        }}>
        <MockedProvider mocks={[studentInfoMock]}>
          <InstitutionSummaryCard institution={institution} />
        </MockedProvider>
      </UserInfoProvider>,
      {
        initialState: { session: { loginError: {}, user: { type: 'student' } } },
      }
    );

    expect(screen.queryByText('Direct application')).toBeInTheDocument();
    expect(screen.queryByText('Common App application')).toBeInTheDocument();

    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(4);
    expect(buttons).toMatchSnapshot();
  });
});
