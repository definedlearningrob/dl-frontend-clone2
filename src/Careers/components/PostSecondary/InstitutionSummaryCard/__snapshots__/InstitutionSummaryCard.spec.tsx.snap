// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstitutionSummaryCard should always display apply buttons 1`] = `
[
  <button
    class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] button !border-neutral-300 hover:!border-danger-600 hover:!text-danger-600 hover:!bg-danger-100 focus-visible:outline focus-visible:!outline-danger-600 focus-visible:!outline-1 focus-visible:!outline-offset-2 !text-danger-600 !border-danger-600 -default bg-white text-font-primary hover:bg-primary-200 outline-primary-500 border-white hover:border-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
    data-testid="button"
    type="button"
  >
    <div
      class="flex"
    >
      <div
        class="flex align-center justify-center grow-0 shrink-0 text-danger-600 h-[14px] mt-[-1px]"
        data-testid="icon"
      >
        <div
          class="h-sm w-sm"
        />
      </div>
    </div>
    <span
      class="items-center flex justify-center gap-xs"
    >
      Favorite
    </span>
  </button>,
  <button
    class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-xs -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
    data-testid="button"
    type="button"
  >
    <div
      class="flex"
    >
      <div
        class="flex align-center justify-center grow-0 shrink-0 h-[14px] mt-[-1px]"
        data-testid="icon"
      >
        <div
          class="h-sm w-sm"
        />
      </div>
    </div>
    <span
      class="items-center flex justify-center gap-xs"
    >
      Website
    </span>
  </button>,
  <button
    class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline py-xs px-sm !text-xs leading-[14px] !gap-xxs border -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
    data-testid="button"
    type="button"
  >
    <span
      class="items-center flex justify-center gap-xs"
    >
      Common App application
    </span>
  </button>,
  <button
    class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline py-xs px-sm !text-xs leading-[14px] !gap-xxs border -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-base leading-[22px] border"
    data-testid="button"
    type="button"
  >
    <span
      class="items-center flex justify-center gap-xs"
    >
      Direct application
    </span>
  </button>,
]
`;

exports[`InstitutionSummaryCard should render correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="container"
    >
      <div
        class="imageWrapper"
      >
        <div
          class="absolute left-base top-base rounded-xs px-xs py-xxs bg-white"
        >
          <div />
        </div>
        <img
          alt="University of Idaho"
          class="image"
          src="image-url"
        />
      </div>
      <div
        class="cardContent"
      >
        <div
          class="header"
        >
          <div>
            <div
              class="tag"
            >
              Moscow, ID
            </div>
            <h2
              class="text-font-primary mb-0 titleMedium title"
              data-testid="card-title"
            >
              University of Idaho
            </h2>
          </div>
          <div
            class="actions"
          >
            <button
              class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] button !border-neutral-300 hover:!border-danger-600 hover:!text-danger-600 hover:!bg-danger-100 focus-visible:outline focus-visible:!outline-danger-600 focus-visible:!outline-1 focus-visible:!outline-offset-2 !text-danger-600 !border-danger-600 -default bg-white text-font-primary hover:bg-primary-200 outline-primary-500 border-white hover:border-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
              data-testid="button"
              type="button"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 text-danger-600 h-[14px] mt-[-1px]"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              <span
                class="items-center flex justify-center gap-xs"
              >
                Favorite
              </span>
            </button>
            <a
              href="https://www.uidaho.edu"
              target="_blank"
            >
              <button
                class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-xs -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
                data-testid="button"
                type="button"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 h-[14px] mt-[-1px]"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  class="items-center flex justify-center gap-xs"
                >
                  Website
                </span>
              </button>
            </a>
          </div>
        </div>
        <div
          class="description"
        >
          <p
            class="mb-base xxxl:mb-md"
          >
            University of Idaho is a medium, Public, 4-year or above institution. University of Idaho is located in distant town. It offers certificates, bachelor's degrees, master's degrees, doctoral degrees.
          </p>
          <div
            class=""
          >
            <p>
              Begin application process
              :
            </p>
            <div
              class="flex gap-xs xxxl:gap-sm"
            >
              <a
                href="https://url-for-application"
                target="_blank"
              >
                <button
                  class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline py-xs px-sm !text-xs leading-[14px] !gap-xxs border -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
                  data-testid="button"
                  type="button"
                >
                  <span
                    class="items-center flex justify-center gap-xs"
                  >
                    Common App application
                  </span>
                </button>
              </a>
              <a
                href="https://www.uidaho.edu"
                target="_blank"
              >
                <button
                  class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline py-xs px-sm !text-xs leading-[14px] !gap-xxs border -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-base leading-[22px] border"
                  data-testid="button"
                  type="button"
                >
                  <span
                    class="items-center flex justify-center gap-xs"
                  >
                    Direct application
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
